### PR TITLE
Usage with Expo instructions

### PR DESCRIPTION
**Pull request by peterpme**

### CODE REVIEW

**Comment by adam-stasiak** [comment]
Hey, Could you add please an instruction for Android setup? I have already raised an issue #578 and my team cannot proceed with android tests.

File: docs/Guide.Expo.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      "binaryPath": "bin/Exponent.app",
+      "type": "ios.simulator",
+      "name": "iPhone 7"
+    }
+  }
+}
+```
+
+- Download the Expo app from [Expo.io/tools](https://expo.io/tools). As of 03/19/18
+  - [iOS IPA 2.3.0](https://dpq5q02fu5f55.cloudfront.net/Exponent-2.3.0.tar.gz)
```
Comment (rotemmiz) [inline docs/Guide.Expo.md]: Try being less specific here, we don't want to update this regularly. Just put a link to the download page (https://expo.io/tools)

File: docs/Guide.Expo.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - [Android APK 2.3.2](https://d1ahtucjixef4r.cloudfront.net/Exponent-2.3.2.apk)
+
+- Unzip the iOS IPA and **rename the folder** to `Exponent.app` (it'll now be a file and not a folder)
```
Comment (rotemmiz) [inline docs/Guide.Expo.md]: It will now have a file icon (though is still a folder)

**Comment by peterpme** [comment]
Thanks @rotemmiz!

**Comment by rotemmiz** [comment]
Hey @peterpme , 

Can you please relate to both platforms ? even if Android is not currently working, or you're not sure how to, just mention it in docs, se we'd take it from there.

**Comment by peterpme** [comment]
I added a TBD section for Android

**Comment by rotemmiz** [comment]
Please rephrase the Android part, it seems more like a personal note than documentation. 😯

File: docs/Guide.Expo.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+That's it! The rest of what you do should be similar to normal detox settings. If you have questions, [tweet @peterpme](https://twitter.com/peterpme)
+
+## Usage with Expo (Android)
+
+- We haven't personally tried getting this to work on Android. If you have, feel free to open up a PR!
+
+### Example App
+The [example app](https://github.com/expo/with-detox-tests) is outdated but the code is exactly the same. I have a PR open that'll make it runnable again.
```
Comment (rotemmiz) [inline docs/Guide.Expo.md]: This is the line I was referring to

**Comment by peterpme** [comment]
:wave: @rotemmiz anything else?

**Comment by rotemmiz** [comment]
Hey @peterpme , looks like your PR was accepted in expo, right?
Please change the note.

**Comment by DanielMSchmidt** [comment]
@peterpme Any updates here? 😇

**Comment by peterpme** [comment]
@DanielMSchmidt how's that? Thanks!

File: docs/Guide.Expo.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- Create `bin` folder in your project and put `Exponent.app` inside so it matches the binaryPath you set above.
+- Create an `e2e` and copy over the settings from [the example app](https://github.com/expo/with-detox-tests/tree/master/e2e)
+
+The most important piece of this the `reloadApp` from `detox-expo-helpers`. Don't forget this.
+
+```es6
+const { reloadApp } = require('detox-expo-helpers');
+// ...
+beforeEach(async () => {
+  await reloadApp();
```
Comment (bmakuh) [inline docs/Guide.Expo.md]: Out of curiosity, is it possible at all to _not_ have to reload the app in between every test? The setup/teardown process adds a fairly significant amount of time to test runs, but when I remove this beforeEach hook from my tests the tests get screwy, so I was wondering if this is just the way it is for now.

**Comment by CodingBauer** [comment]
Hi @peterpme, thanks a lot for this from a CRNA user!
I am getting close, but it still seems that the tests just time out. Is maybe some step missing?

File: docs/Guide.Expo.md (hunk 4 above)
Comment (peterpme) [reply docs/Guide.Expo.md]: I think this is just the way it has to be for now.

**Comment by peterpme** [comment]
Hey @CodingBauer I'm not entirely sure what your issues can be. It's been a couple months since I've opened the PR and a bunch may have changed 😅

**Comment by rotemmiz** [comment]
Hey @peterpme,
Sorry for getting back to this after such a long time, I wasn't sure the addition was precise enough, therefore didn't merge it back then, but on the other hand didn't communicate my doubts, and I am sorry for that. 
So, to fix what's I have broken, let's discuss: 
1. Since things might have change (or already have been changed), and maybe even because of that, try minimizing any expo related explanation and setup (e.g, bin download), instead use links, like you already have.
2. Try keeping the text professional. Don't use  "The most important piece of this" or "don't forget". I know we do it some places ourselves (and we shouldn't), but we want to make high quality docs.
3. No need to mention specific twitter accounts or users.

I would gladly accept this docs addition, if you can please help aligning the text to these guidelines.

BTW, I think it would be a good idea to actually post these, and maybe a few more, guidelines to contribution guide (we need specific guidelines for each part I guess: docs, cross platform features, sync issues, tests, etc).

**Comment by peterpme** [comment]
You got it @rotemmiz!!

I'll make these adjustments over the next couple of days and we can try and get this in :)

Thanks for taking the time to write that out ❤️

**Comment by peterpme** [comment]
The changes were actually quick enough that I just took care of them.

I don't know how to explain this without keeping the instructions to download the `bin` files though. 

Those steps were really helpful for me when I figured them out and if I could save a developer 15-30 
minutes from scratching their heads and wondering how to do it, I think its worth it in the documentation.

Either way - entirely up to you! Thanks again!

**Comment by srikanthpusapatiUnt** [comment]
Can you update this file possibly with how to use Detox using Expo on an Android device?
Thanks.

**Pull request merged**